dynamics/src/include/DynamicsKernel.hpp

```diff
@@ -32,14 +33,20 @@
 namespace Nextsim {
 
 // forward define the class holding the potentially non-DG parts
-template <int DGdegree> class DynamicsInternals;
+template<int DGdegree> class DynamicsInternals;
 
-template <int DGadvection, int DGstress> class DynamicsKernel {
+template<int DGadvection, int DGstress> class DynamicsKernel {
 public:
     typedef std::pair<const std::string, const DGVector<DGadvection>&> DataMapping;
     typedef std::map<typename DataMapping::first_type, typename DataMapping::second_type> DataMap;
 
-    DynamicsKernel() = default;
+    DynamicsKernel(double cosOceanAngleIn, double sinOceanAngleIn, const DynamicsParameters& paramsIn)
+        : cosOceanAngle(cosOceanAngleIn)
+        , sinOceanAngle(sinOceanAngleIn)
+        , baseParams(paramsIn)
+        , deltaT(0.)
```

Comment:
Since ocean_turning_angle is always in DynamicsParameters, wouldn't 

```suggestion
    DynamicsKernel(const DynamicsParameters& paramsIn)
        : cosOceanAngle(std::cos(paramsIn.ocean_turning_angle))
        , sinOceanAngle(std::sin(paramsIn.ocean_turning_angle))
        , baseParams(paramsIn)
        , deltaT(0.)
```

work? This gives fewer arguments for DynamicsKernel and the derived classes.

Reply:
Not so long as `VPCGDynamicsKernel` assigns fixed values to the cos and sin terms, as in
```
    VPCGDynamicsKernel(StressUpdateStep<DGadvection, DGstressComp>& stressStepIn,
        const DynamicsParameters& paramsIn)
        // Note that the ocean turning angle is explicitly zero at present
        : CGDynamicsKernel<DGadvection>(1., 0., paramsIn, u, v)
```
There's no other way (that I know of) to ensure these values are set. Once the `VPCGDynamicsKernel` calculation of stress correctly handles the ocean turning angle, then what you suggest can be done.

Reply:
Ok, I see. It's not difficult to add ocean turning into ``VPCGDynamicsKernel``, but that should probably be a different issue. I'll make one and merge this.